GameServer/Logic/Maps/SummonersRift.cs

```diff
@@ -325,35 +327,117 @@ public override bool Spawn()
                 MinionSpawnPosition.SPAWN_RED_BOT,
                 MinionSpawnPosition.SPAWN_RED_MID,
             };
-
-            if (_waveNumber < 3)
+            if (_game.GetMap().GetSpawnState())
             {
-                for (var i = 0; i < positions.Count; ++i)
+                if (_waveNumber < 3) //First 3 melee minions spawn whatever situation is
```

Comment:
IMO this should be renamed to `_minionNumber` or `_subwaveNumber`, as `_waveNumber` sounds like it refers to the actual minion wave count, which it is not.